x-pack/plugin/deprecation/qa/common/build.gradle

```diff
@@ -0,0 +1,13 @@
+import org.elasticsearch.gradle.util.GradleUtils
+import org.elasticsearch.gradle.internal.info.BuildParams
+
+apply plugin: 'elasticsearch.build'
+tasks.named("test").configure { enabled = false }
+
+dependencies {
```

Comment:
@mark-vieira would you be able to review this PR from gradle point of view?
I wasn't too sure if I should use api or implementation for dependency? do we have a readme for this?
also do I have to copy license files here as well?

Reply:
Licenses are only required once, you don't need them _every_ time you use a dependency. We favor `implementation` as those dependencies are not transitively inherited by the compile classpath of consuming projects. Since this is a QA project it's mostly irrelevant (there are no consumers) but I'd still leave it all `implementation` here.

Reply:
if I remove the licence folder I get 
```
Execution failed for task ':x-pack:plugin:deprecation:qa:common:dependencyLicenses'.
> Licences dir /Users/przemek/workspace/pgomulka/elasticsearch/x-pack/plugin/deprecation/qa/common/licenses does not exist, but there are dependencies: jackson-annotations-2.10.4.jar
  jackson-databind-2.10.4.jar
```

Reply:
Ah, if the build complains then it needs to be there.

Reply:
You shouldn't be adding the `elasticsearch.build` plugin to QA projects. That's specifically for production code. What is this "common" project for anyhow? I don't see any actual code belonging to it?

Reply:
> What is this "common" project for anyhow? 


it has only one class `DeprecationTestUtils` - I felt that copying that code would be a smell.
The problem boils down to the fact that `EarlyDeprecationIndexingIT` has to be first running against a fresh test cluster.

I removed the `elasticsearch.build` but I had to add `elasticsearch.java` otherwise gradle would complain about `implementation` 
```
Could not find method implementation() for arguments
```

Reply:
Yep, `elasticsearch.java` is the correct choice here.